hushline/premium.py

```diff
@@ -0,0 +1,609 @@
+import asyncio
+import json
+from datetime import datetime
+from typing import Tuple
+
+import stripe
+from flask import (
+    Blueprint,
+    Flask,
+    abort,
+    current_app,
+    flash,
+    jsonify,
+    redirect,
+    render_template,
+    request,
+    session,
+    url_for,
+)
+from sqlalchemy import desc
+from werkzeug.wrappers.response import Response
+
+from .db import db
+from .model import (
+    StripeEvent,
+    StripeEventStatusEnum,
+    StripeInvoice,
+    StripeInvoiceStatusEnum,
+    StripeSubscriptionStatusEnum,
+    Tier,
+    User,
+)
+from .utils import authentication_required
+
+FREE_TIER = 1
+BUSINESS_TIER = 2
+
+
+def init_stripe() -> None:
+    stripe.api_key = current_app.config["STRIPE_SECRET_KEY"]
```

Comment:
Just noting that is pattern breaks Flask's app isolation and we're leaking configs between apps. This doesn't matter yet because of our current architecture, but just noting somehow so that if it ever causes problems at least one person other than me has been nudged toward thinking about this.